packages/test-framework/src/fixture/pageObject/Checkout.ts

```diff
@@ -86,4 +86,14 @@ export class Checkout {
     async placeOrder(): Promise<void> {
         await this.page.locator('id=checkout-payment-continue').click();
     }
+
+    async applyCoupon(couponCode: string): Promise<void> {
+        await this.page.locator('[data-test="cart"] [data-test="redeemable-label"]').click();
+        await this.page.locator('[data-test="redeemableEntry-input"]').fill(couponCode);
+        await this.page.locator('[data-test="redeemableEntry-submit"]').click();
+    }
+
+    async removeCoupon(): Promise<void> {
+        await this.page.locator('[data-test="cart-price-callback"]').click();
+    }
```

Comment:
Is this still working when there are multiple coupons?

Reply:
Yes, its working because at a time it allows to only add one coupon as per coupon UI.

Reply:
Becuase this method won't work if a store allows applying multi coupons at a time, I recommend that we leave it out of the test framework and instead use `await this.page.locator('[data-test="cart-price-callback"]').click();`.In the test directly.